Loathing-Associates-Scripting-Society/autoscend
# Support Lucky! adventures

# Description

This is a big one. Changed all uses of Ten-Leaf Clovers and semirares. Either converted to a Lucky! adventure or removed if low value. Added Astral Energy Drink as a Lucky! source. Did not add pillkeeper, lucky lindy, or optimal dog. Pillkeeper wasn't used anywhere to force a semirare.

Kept these clover uses:
1. Power leveling only for You, Robot
2. Level 8 Mine. Note this should consume up to 3
3. Boo Peak
4. Zeplin Protesters if have >= 10 progress potential
5. Wand of Nagamar

Kept these semi rares:
1. Hidden temple for stone wool
2. Top floor of castle for inhaler

Fixes #995 

## How Has This Been Tested?

Running on an alt now. Will take a while to fully test.

Stuf to test in any path:
- [x] Buys clovers as needed throughout the day
- [x] If any clovers left at bedtime, buy all
- [x] Pull a clover at day end if in softcore and haven't pulled one yet today
- [x] Hidden Temple for Stone Wool x2
- [x] Level 8 mine for ores. Goal is to not get outfit at all
- [x] Boo Peak for clue x2
- [x] Top floor of Castle for inhaler if doing Nuns
- [x] Zeplin protesters if >=10 progress can be made
- [ ] Wand of Nagamar

Nice to have tests
- Run in BHY
- Run in Glover
- Run in Plumber. There was a check in the old semirare handler for plumber that I don't understand.

## Checklist:

- [x] I have based by pull request against the [master branch](https://github.com/Loathing-Associates-Scripting-Society/autoscend/tree/master) or have a good reason not to.
